diff.go

```diff
@@ -0,0 +1,193 @@
+package cidrman
+
+import (
+	"fmt"
+	"net"
+)
+
+// DiffIPNets accepts two lists of mixed IP networks and return three lists of
+// IPNets that only exists in first list, second lost and in both lists, together with a status number.
+// The DiffIPNets() will return the smallest possible lists of IPNets.
+// Example:
+//    onlyInLeftNets, inBothNets, onlyInRightNets, status, err := DiffIPNets(leftListOfNets, rightListOfNets)
+// status values:
+//    0 = both lists with exact same nets after individual merge
+//    1 = leftnets and rightnets differ with some overlap and some unique nets
+//    2 = leftnets same as rightnets with some extra nets
+//    3 = rightnets same as leftnets with some extra nets
+//    4 = leftnets and rightnets had no overlap at all
+//    5 = empty input nets
+//
+//   -1 = some error processing nets in sub functions, see err
+//   -2 = missing input data, see err
+func DiffIPNets(leftnets, rightnets []*net.IPNet) (leftunique, inboth, rightunique []*net.IPNet, status int, err error) {
+	if leftnets == nil || rightnets == nil {
+		return nil, nil, nil, -2, fmt.Errorf("Missing input data in DiffINets()")
+	}
+	if len(leftnets) == 0 && len(rightnets) == 0 {
+		return make([]*net.IPNet, 0), make([]*net.IPNet, 0), make([]*net.IPNet, 0), 5, nil
+	}
+	if len(leftnets) == 0 {
+		return make([]*net.IPNet, 0), make([]*net.IPNet, 0), rightnets, 3, nil
+	}
+	if len(rightnets) == 0 {
+		return leftnets, make([]*net.IPNet, 0), make([]*net.IPNet, 0), 2, nil
+	}
+
+	// Merge nets and subsetnets individually to have the miminal set of largets networks
+	leftnets, err = MergeIPNets(leftnets)
+	if err != nil {
+		return nil, nil, nil, -1, fmt.Errorf("Merge error leftnets: %w", err)
+	}
+	rightnets, err = MergeIPNets(rightnets)
+	if err != nil {
+		return nil, nil, nil, -1, fmt.Errorf("Merge error rightnets: %w", err)
+	}
+
+	// Split into IPv4 and IPv6 lists.
+	// Handle the list separately.
+	var left4s cidrBlock4s
+	var left6s cidrBlock6s
+	for _, net := range leftnets {
+		ip4 := net.IP.To4()
+		if ip4 != nil {
+			left4s = append(left4s, newBlock4(ip4, net.Mask))
+		} else {
+			ip6 := net.IP.To16()
+			left6s = append(left6s, newBlock6(ip6, net.Mask))
+		}
+	}
+	var right4s cidrBlock4s
+	var right6s cidrBlock6s
+	for _, net := range rightnets {
+		ip4 := net.IP.To4()
+		if ip4 != nil {
+			right4s = append(right4s, newBlock4(ip4, net.Mask))
+		} else {
+			ip6 := net.IP.To16()
+			right6s = append(right6s, newBlock6(ip6, net.Mask))
+		}
+	}
+
+	// Create leftunique
```

Comment:
Below we are doing the same things twice, is it possible to easily refactor the code so that `removeX` handles the zero-length case and generalize the function? I.e. something like:
```
leftunique = append(remove(left4s, right4s), ...)
```
where `remove` has the smartness to decide to call on 4/6 version of helper funcs?

Reply:
It would be possible to use RemoveIPNets() from remove.go and SubsetIPNets() from subset.go. I'll look into it.